editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -53,15 +53,22 @@ static void _draw_shadowed_line(Control *p_control, const Point2 &p_from, const
 	p_control->draw_line(p_from + p_shadow_offset, p_from + p_size + p_shadow_offset, p_shadow_color);
 }
 
-void SpriteFramesEditor::_open_sprite_sheet() {
-	file_split_sheet->clear_filters();
-	List<String> extensions;
-	ResourceLoader::get_recognized_extensions_for_type("Texture2D", &extensions);
-	for (int i = 0; i < extensions.size(); i++) {
-		file_split_sheet->add_filter("*." + extensions[i]);
-	}
+void SpriteFramesEditor::_open_sprite_sheet(bool p_force_open) {
+	//If there's no texture registered then open the popup
+	if (EditorSettings::get_singleton()->get_project_metadata("sprite_frames", "last_texture", "") == "" || Input::get_singleton()->is_key_pressed(Key::SHIFT) || p_force_open) {
```

Comment:
```suggestion
	if (p_force_open || Input::get_singleton()->is_key_pressed(Key::SHIFT) || EditorSettings::get_singleton()->get_project_metadata("sprite_frames", "last_texture", "") == "") {
```
Put the cheapest conditions first